backend/apps/volontulo/views/api.py

```diff
@@ -65,6 +69,54 @@ def login_view(request):
     )
 
 
+@api_view(['POST'])
+@authentication_classes((CsrfExemptSessionAuthentication,))
+@permission_classes((AllowAny,))
+def register_view(request):
+    """REST API register view."""
+    if request.user.is_authenticated():
+        return Response(status=status.HTTP_400_BAD_REQUEST)
+
+    email = request.data.get('email')
+    password = request.data.get('password')
+
+    user = User.objects.create_user(
+        username=email,
+        email=email,
+        password=password,
+        is_active=False,
+    )
+    try:
+        user.save()
+    except IntegrityError:
+        return Response(status=status.HTTP_201_CREATED)
+
+    profile = UserProfile(user=user)
+    ctx = {'token': profile.uuid}
+    profile.save()
+    send_mail(request, 'registration', [user.email], context=ctx)
+
+    return Response(status=status.HTTP_201_CREATED)
+
+
+
+@api_view(['POST'])
+@authentication_classes((CsrfExemptSessionAuthentication,))
+@permission_classes((AllowAny,))
+def activate_view(request, uuid):
+    """View responsible for activating user account."""
+    try:
+        profile = UserProfile.objects.get(uuid=uuid)
```

Comment:
It seems that this activation token is valid forever, isn't it, @magul? I don't think it's acceptable from the security point of view. Anyway, it's been like that before as well, so it's not an argument to reject this PR.

Reply:
It was that way before and I don't recognize any security issue here, as already used token is useless (doesn't do anything).